### ISSUE TITLE
bug fixing / secondary implementation fixes / Testing coverage (cleaning and seasoning the meat and potatoes)

### ISSUE THREAD

**Issue by mtcA6** [department-of-veterans-affairs/va.gov-team#61013]

**Comment by adamwhitlock1** [comment]
PR is in review, has one approval, although since it is a large PR overall, I am going to wait until tomorrow to merge and see if any other feedback comes in.

I'd push this one into next sprint as a 1ish point of remaining work.

**Comment by adamwhitlock1** [comment]
This has been deployed to staging and is ready to validate. I flipped the flags for checkboxes and email as a channel for all users in staging for now to make it easier to look at the feature with different users and edge cases.

I guess steps for validating would be to : 
- log in as a user and view checkboxes and see that they work
- spacing/margins appear to be in line with designs
- remove email from contact information and verify that the expandable alert appears in notification settings
- remove mobile from contact information and verify that the expandable alert appears in notification settings
- verify that default send indicator works as expected. I'll have to look at some specific users and see how that can be validated specifically, so I'll get back to this one and add some more details. I validated locally with mock data, but definitely want to have a set of test users that can be used on staging for checking this feature out.

**Comment by mtcA6** [comment]
I think everything is looking good. One thing i noticed however is the MHV newsletter is under "general VA updates and info" while i don't disagree...

![](https://github.com/department-of-veterans-affairs/va.gov-team/assets/129431463/2117ae16-d898-4f16-b9ad-0942a632d3a3)

...the mocks show it as under healthcare: 

![](https://github.com/department-of-veterans-affairs/va.gov-team/assets/129431463/d6c71796-6df3-46f1-bb70-35c9cb936259)

Can we correct that as part of #62077

**Comment by mtcA6** [comment]
@Samara-Strauss, Liz reminded me that nomenclature for the heading is coming from VA Profile. Do you think it's worth bringing this up with them and having them update the 'metadata' associated w/those fields or do we handle this in our FE code only? (may not be a technically accurate term but it imparts, i think, the correct meaning here) 

I know technically we can change the heading since we discussed this on a different ticket, but wondering aloud here if we should

**Comment by Samara-Strauss** [comment]
So it sounds like there are two parts to this question, but let me know if I'm not getting at what you're asking:

- If you need to change what category a preference is in, yes, I believe that VA Profile has to change that. In terms of where the newsletter preference goes, @andaleliz was the recommendation from IA to have that in the "Your health care" section? TBH, I think it fits just fine under "General VA updates and information", but we can lead with whatever our content & IA folks recommend.
- As for any metadata associated with those fields, I don't know how that works, @mtcA6. My recommendation would be to ask Adam for specifics.

**Comment by mtcA6** [comment]
I may have been mistaken in how i was understanding the structure of things.

When i was writing [this comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/61013#issuecomment-1649841018) above, I think i confused [this comment about the "notify by text"](https://github.com/department-of-veterans-affairs/va.gov-team/issues/59721#issuecomment-1581288998) or email verbiage and how we explicitly code that in the UI. The headings/headers looks like its returned in the API as the 'communication group' \< i may be misreading that I'm not the best at reading json.   
  
I guess my question remains, do we change it unilaterally in the UI (and can we) to reflect the thing we want or do we necessarily have to go back to VA Profile to get it updated?  \<Taking a next step here is of course dependent on how Liz answers the question about what the Content/IA folks want to see.

**Comment by Samara-Strauss** [comment]
[This documentation is super helpful in quickly showing what we control vs. what VA Profile controls](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/profile/notification-preferences/frontend/display-notes.md#tldr). We do control the "Notify by [email/text]" copy, so we would update that, not VA Profile.

**Comment by adamwhitlock1** [comment]
Yep, so any changes to a notification name, notification description,  group name, or location of a notification in relation to the groups would be handled by asking VA Profile.

While it's technically possible to override any of those things on the FE by writing some code to do such a thing, I would only resort to that as a last ditch effort or temporary fix if absolutely needed, since VA Profile should be the source of truth, and such bandaid changes would introduce tech debt and maintenance burdens.

**Comment by Samara-Strauss** [comment]
> We don't currently have a notification group on the FE called "General VA updates and information", though it might be in the VA Profile code from the MVP

Can confirm this section exists in VA Profile's code, it just hasn't had anything in it until now.

> We'd need to decide if we think the MHV Newsletter is something all Veterans would be interested in even if they don't have health care, since the Your Health section is conditional. I'm not sure exactly who would make that call.

I think we can lead with a strong recommendation. I would think that yes, this probably only really applies to health care users. However, since it is not personalized and doesn't have anything to do with their individual care, I think the "General" section works. But that's just one opinion, so I'd be curious what content & IA folks recommend.

**Comment by adamwhitlock1** [comment]
> @adamwhitlock1 , remind me again, do we have two toggles? since staging is down asking since i can't look
> 
> 1. a toggle for the default send stuff ie. the checkboxes
> 2. a toggle for the mhv notification options

There are a ton of toggles involved in the notification settings page and the various pieces of functionality.

I will update my docs on the toggles involved, but here are the 6 main ones.

- `profile_use_notification_settings_checkbozes`
  - toggles the usage of radio buttons vs checkboxes as the input type that is used for every notification. Also consumes default send indicator boolean when turned on.

- `profile_show_payments_notification_setting`
  - Used to show/hide Payments group and related notifications in Notification Settings page

- `profile_show_email_notification_settings`
  - toggles display of ALL email based notification checkboxes

- `profile_show_mhv_notification_settings`
  - toggles display of ALL MHV notifications
  - the MHV notifications all happen to be email based, so this is similar to the above, but will allow any future email notifications to be toggled separately
  - RX refill shipment notification (needs combined with Prescription shipment and tracking updates via VA Profile)
  - VA Appointment Reminders (needs to be combined with Appointment Reminders via VA Profile)
  - Secure messaging alert
  - Medical images and reports available
  - Biweekly MHV newsletter (in turn the General VA Updates and Information group is hidden since this notification is the only one in the group) 

- `profile_show_quick_submit_notification_setting`
  - should not be turned on at all on prod at this point
  - toggles display of QuickSubmit group and QuickSubmit Upload Status notification

- `profile_use_field_editing_page` - somewhat related
  - allows an 'editing as a sub-task' UI to be used when updating a single section of profile. Used for mobile phone and email address sections via notification settings page alerts when one of those channels is missing from the user's profile AND the `profile_show_email_notification_settings` is turned ON
  
  
So yea you can imagine how happy I will be when we get some of this deployed and we can reduce some of these toggles. I'm sure you can imagine the complexity of what a mix of all these being in various states of ON/OFF can lead to.

They are all ON is staging at this point, except for the QuickSubmit one since that is supposed to just be turned off completely.

**Comment by mtcA6** [comment]
@adamwhitlock1 thanks for the details here, this is immensely helpful for writing the combined release plan for editing as a subtask and email as a channel ahead of staging review for both things.

**Comment by Samara-Strauss** [comment]
@adamwhitlock1 this is so helpful! [I created this doc and linked it from the main product outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/profile/notification-preferences/frontend/feature-flags.md).
